compute/cker/include/cker/train/operation/MaxPool.h

```diff
@@ -0,0 +1,131 @@
+/*
+ * Copyright (c) 2023 Samsung Electronics Co., Ltd. All Rights Reserved
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef __NNFW_CKER_TRAIN_OPERATION_MAXPOOL_H__
+#define __NNFW_CKER_TRAIN_OPERATION_MAXPOOL_H__
+
+#include "cker/Shape.h"
+#include "cker/Utils.h"
+#include "cker/eigen/Utils.h"
+
+#include <Eigen/Core>
+
+namespace nnfw
+{
+namespace cker
+{
+namespace train
+{
+
+// Most of the logic except 'arg_max_index' related is copy-paste from
+// https://github.com/Samsung/ONE/blob/a380292/compute/cker/include/cker/operation/MaxPool.h#L42-L88
+// 'arg_max_index' is to record max-arguments' index to apply gradient later.
```

Comment:
### To reviewers : 

I personally think this code is a bit hard to read, So I'd like to help reviewing. But I don't know how to. :thinking:  Here is some comments to give you the context! 

-------------------------------- 

* As the comment mentioned, Most of the code is from MaxPool Inference kernel : https://github.com/Samsung/ONE/blob/a380292/compute/cker/include/cker/operation/MaxPool.h#L42-L88. 

* Only difference is the `max-arg-index`. It records the max-agruments' index while forwarding. ( https://github.com/Samsung/ONE/pull/11666/files#r1348234693 ) And while backwarding, it passes the gradient to the recorded index. 

* Test code might help to understand : https://github.com/Samsung/ONE/pull/11666/files#r1348220649 

Please feel free to ask anything! I'm open to any other idea :smile: